src/app/view/add-device-group/add-device-group.component.ts

```diff
@@ -360,12 +382,20 @@ export class AddDeviceGroupComponent {
     this.displayList(this.p);
   }
   displayList(page: number) {
-    this.deviceservice
-      .getfilterData(this.FilterForm.value, this.orgId, page)
-      .subscribe((data) => {
-        this.loading = false;
-        this.updateDeviceTable(data.devices, data.totalCount, data.totalPages);
-      });
+    if (this.selectionType === SelectionType.Checkbox) {
+      this.deviceservice
```

Comment:
this part seems to be repeated multiple times, can you instead convert it into a reusable function